core/commons/che-core-commons-observability/src/main/java/org/eclipse/che/commons/observability/ExecutorServiceWrapper.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.commons.observability;
+
+import com.google.common.annotations.Beta;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ScheduledExecutorService;
+import org.eclipse.che.commons.schedule.executor.CronExecutorService;
+
+/**
+ * Wrapper of different implemntations of {@link ExecutorService}. At this moment supported: {@link
+ * ExecutorService}, {@link ScheduledExecutorService} amd {@link CronExecutorService}.
+ *
+ * <p>Depending on implementation and environment configuration wrapper can add to the original
+ * class different capabilities like monitoring or tracing.
+ *
+ * @author Sergii Kabashniuk
+ */
+@Beta
+public interface ExecutorServiceWrapper {
+
+  /**
+   * Creates wrapper for the given executor.
+   *
+   * @param executor - {@link ExecutorService} that has to be wrapped.
```

Comment:
ASAIR no needs in `-` in java docs. Use Preview Java Docs feature in IDE to check what you'll get at the end.